### PR TITLE
v5.1

### PR DESCRIPTION
**Pull request by jbender**
Merges in a bunch of upstream fixes, and starts to get specs running again

### CODE REVIEW

**Comment by jbender** [comment]
@rnaud / @nickelser This should be good to go, and for the first time, with passing specs to prove it.

https://travis-ci.org/jbender/odbc_adapter/builds/493921681

BTW, I don't have sufficient permissions to push directly/enable the test suite on Travis for this repo, but we should probably do that then tie in passing builds to PRs

File: lib/odbc_adapter/adapters/mysql_odbc_adapter.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class MySQLODBCAdapter < ActiveRecord::ConnectionAdapters::ODBCAdapter
       PRIMARY_KEY = 'INT(11) NOT NULL AUTO_INCREMENT PRIMARY KEY'.freeze
 
       class BindSubstitution < Arel::Visitors::MySQL
-        include Arel::Visitors::BindVisitor
+        include Arel::Visitors::BindVisitor if Arel::VERSION.to_i < 9
```
Comment (nickelser) [inline lib/odbc_adapter/adapters/mysql_odbc_adapter.rb:9]: might be good to have a comment on this restriction

File: .travis.yml [hunk 2]
```diff
@@ -1,29 +1,43 @@
...
+  - gemfiles/active_record_5_1.gemfile
+  - gemfiles/active_record_5_2.gemfile
+
+before_script:
+  - sh -c "if [ '$DB' = 'mysql' ]; then sudo odbcinst -i -d -f /usr/share/libmyodbc/odbcinst.ini; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then mysql -e 'DROP DATABASE IF EXISTS odbc_test; CREATE DATABASE IF NOT EXISTS odbc_test;' -uroot; fi"
+  - sh -c "if [ '$DB' = 'mysql' ]; then mysql -e 'GRANT ALL PRIVILEGES ON *.* TO "root"@"localhost";' -uroot; fi"
+
+  - sh -c "if [ '$DB' = 'postgresql' ]; then sudo odbcinst -i -d -f /usr/share/psqlodbc/odbcinst.ini.template; fi"
+  - sh -c "if [ '$DB' = 'postgresql' ]; then psql -c 'CREATE DATABASE odbc_test;' -U postgres; fi"
```
Comment (nickelser) [inline .travis.yml:43]: ideally we would have some test account w/snowflake but i see that being somewhat difficult (and definitely not in scope here :)

File: lib/odbc_adapter/database_statements.rb [hunk 3]
```diff
@@ -128,7 +128,7 @@ def nullability(col_name, is_nullable, nullable)
     end
 
     def prepared_binds(binds)
-      prepare_binds_for_database(binds).map { |bind| _type_cast(bind) }
+      binds.map(&:value_for_database).map { |bind| _type_cast(bind) }
```
Comment (nickelser) [inline lib/odbc_adapter/database_statements.rb:131]: i'm not following this change here -- what's wrong with the `prepare_binds_for_database_method`?

**Review by nickelser — APPROVED** [review]
overall this looks great, with some changes that don't fully make sense to me (but don't look wrong either 😄)

File: lib/odbc_adapter/database_statements.rb (hunk 3 above)
Comment (jbender) [reply lib/odbc_adapter/database_statements.rb:131]: I don't remember exactly what prompted it, but it's an internal method that's been removed from Rails:

https://github.com/rails/rails/commit/fb30df9c10372b1667dabac2fffe2cff25302626

**Comment by jbender** [comment]
@nickelser I cleaned up the history and split this hydra into their respective parts (adding a comment about the Arel issue). If you want to go ahead and merge those, I'd recommend doing them in the order that I added the PRs.

**Comment by nickelser** [comment]
sounds good, thanks for doing this! will merge now

**Pull request closed without merging**